### PR TITLE
[Validator] Deprecate annotations in favor of attributes

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | yes
| Tickets       | Follows #50888
| License       | MIT
| Doc PR        | symfony/symfony-docs#18589

This PR deprecates using Doctrine annotations to configure validation constraints. Attributes shall be used instead.

Existing applications can be migrated easily using [Rector](https://getrector.com/blog/how-to-upgrade-annotations-to-attributes).

Deprecation errors triggered by the bundles' functional tests will be resolved once #50888 is merged.

### CODE REVIEW

File: src/Symfony/Component/Validator/Tests/Mapping/Loader/AnnotationLoaderWithHybridAnnotationsTest.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+use Doctrine\Common\Annotations\AnnotationReader;
+use Symfony\Bridge\PhpUnit\ExpectDeprecationTrait;
+use Symfony\Component\Validator\Mapping\Loader\AnnotationLoader;
+
+class AnnotationLoaderWithHybridAnnotationsTest extends AnnotationLoaderTest
+{
+    use ExpectDeprecationTrait;
+
+    /**
+     * @group legacy
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Tests/Mapping/Loader/AnnotationLoaderWithHybridAnnotationsTest.php]: can be moved at the class level maybe?

File: src/Symfony/Component/Validator/Tests/Mapping/Loader/AnnotationLoaderWithLegacyAnnotationsTest.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+use Doctrine\Common\Annotations\AnnotationReader;
+use Symfony\Bridge\PhpUnit\ExpectDeprecationTrait;
+use Symfony\Component\Validator\Mapping\Loader\AnnotationLoader;
+
+class AnnotationLoaderWithLegacyAnnotationsTest extends AnnotationLoaderTest
+{
+    use ExpectDeprecationTrait;
+
+    /**
+     * @group legacy
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Tests/Mapping/Loader/AnnotationLoaderWithLegacyAnnotationsTest.php]: class level?

**Review by nicolas-grekas — COMMENTED** [review]
please mind fabbot, it has useful hints

**Comment by derrabus** [comment]
> please mind fabbot, it has useful hints

The fact that it does not want me to fully qualify a `method_exists()` call always puzzles me. 🤷🏻‍♂️

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**